prometheus-metrics-model/src/main/java/io/prometheus/metrics/model/registry/PrometheusRegistry.java

```diff
@@ -72,6 +72,29 @@ public MetricSnapshots scrape() {
         return result.build();
     }
 
+    public MetricSnapshots scrape(PrometheusScrapeRequest scrapeRequest) {
```

Comment:
There's a lot of duplicate code in `PrometheusRegistry` now. Maybe it makes sense to implement the existing `scrape()` methods like this:

```java
    public MetricSnapshots scrape() {
        return scrape((PrometheusScrapeRequest) null);
    }

    public MetricSnapshots scrape(Predicate<String> includedNames) {
        return scrape(includedNames, null);
    }
```

And refactor the implementations of the new `scrape()` methods to call `collect()` like this:

```java
scrapeRequest == null ? collector.collect() : collector.collect(scrapeRequest);
```